src/doc/unstable-book/src/language-features/macro-metavar-expr-concat.md

```diff
@@ -0,0 +1,154 @@
+# `macro_metavar_expr_concat`
+
+The tracking issue for this feature is: [#124225]
+
+------------------------
+
+
+`#![feature(macro_metavar_expr_concat)]` provides a more powerful alternative to [`concat_idents!`].
+
+> This feature is not to be confused with [`macro_metavar_expr`] or [`concat_ident`].
+
+> This is an experimental feature; it and its syntax will require a RFC before stabilization.
+
+
+### Overview
+
+`macro_rules!` macros cannot create new identifiers and use them in ident positions.
```

Comment:
I'm not sure about the "in ident position" phrasing. Is there a better (or maybe stronger/weaker) way of describing where it can be used?

Reply:
It's fine to say that we can't create identifiers at all, since this isn't possible on stable. This section could be something like:

> In stable Rust, there is no way to create new identifiers by joining identifiers to literals or other identifiers without using procedural macros such as `paste`. `#![feature(macro_metavar_expr_concat)]` introduces a way to do this, using the `concat` metavariable expression:
>
> ```rust
> // example
> ```
>
> The above expands to:
>
> ```rust
> // expansion
> ```

Reply:
I've used this wording at the top but kept this section as is, I prefer a "show a locked door before you show a key" approach.